sqlglot/transforms.py

```diff
@@ -296,19 +296,41 @@ def unqualify_unnest(expression: exp.Expression) -> exp.Expression:
     return expression
 
 
-def unnest_to_explode(expression: exp.Expression) -> exp.Expression:
+def unnest_to_explode(
+    expression: exp.Expression,
+    unnest_using_arrays_zip: bool = True,
+    support_unnest_func: bool = False,
+) -> exp.Expression:
     """Convert cross join unnest into lateral view explode."""
+
+    def handle_unnest(u: exp.Unnest, _has_multi_expr: bool) -> None:
+        if unnest_using_arrays_zip and _has_multi_expr:
+            # Use INLINE(ARRAYS_ZIP(...)) for multiple expressions
+            u.set(
+                "expressions",
+                [exp.Anonymous(this="ARRAYS_ZIP", expressions=u.expressions)],
+            )
+
+    def udtf_type(u: exp.Unnest, _has_multi_expr: bool):
+        return (
+            exp.Posexplode
+            if u.args.get("offset")
+            else (exp.Inline if unnest_using_arrays_zip and _has_multi_expr else exp.Explode)
+        )
+
     if isinstance(expression, exp.Select):
         from_ = expression.args.get("from")
 
-        if from_ and isinstance(from_.this, exp.Unnest):
+        if not support_unnest_func and from_ and isinstance(from_.this, exp.Unnest):
             unnest = from_.this
             alias = unnest.args.get("alias")
-            udtf = exp.Posexplode if unnest.args.get("offset") else exp.Explode
+            has_multi_expr = len(unnest.expressions) > 1
+            handle_unnest(unnest, has_multi_expr)
             this, *expressions = unnest.expressions
+
```

Comment:
Note that the `Expression` properties such as `expr.this`, `expr.expressions` etc are dict lookups under the hood so it's best if we can cache the result; Here `unnest.expressions` is called on both line 327 and 329, so we can either cache `unnest.expressions` or use `*expressions` as a proxy for `has_multi_expr`

Reply:
We set the default alias in `starrocks.py`:
```
unnest.set("alias", alias)
```
The following `starrocks sql` will use this logic:
```
SELECT student, score, unnest FROM tests CROSS JOIN LATERAL UNNEST(scores)
```
But I found that ` join_expr.args.get("alias")` may not get alias, it is only in Unnest, `sqlglot.dialects.starrocks.StarRocks.Parser._parse_unnest` can not set the lateral alias, am I missing something?
I provide details:

```
Select(
  expressions=[
    Column(
      this=Identifier(this=student, quoted=False)),
    Column(
      this=Identifier(this=score, quoted=False)),
    Column(
      this=Identifier(this=unnest, quoted=False))],
  from=From(
    this=Table(
      this=Identifier(this=tests, quoted=False))),
  joins=[
    Join(
      this=Lateral(
        this=Unnest(
          expressions=[
            Column(
              this=Identifier(this=scores, quoted=False))],
          alias=TableAlias(
            columns=[
              Identifier(this=unnest, quoted=False)]))),
      kind=CROSS)])
```